pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py

```diff
@@ -25,7 +25,7 @@ def __init__(
             batch_size: int = 32,
             learning_rate: float = 0.001,
             data_dir: str = '.',
-            datamodule: pl_bolts.datamodules.LightningDataModule = None,
+            datamodule: LightningDataModule = None,
```

Comment:
@nateraw does trainer.fit(ae) automatically accept self.datamodule in this case? I can't see a call to trainer.fit(ae, datamodule)

Reply:
If you set `self.datamodule` in a `LightningModule`, `train_dataloader`, `val_dataloader`, and `test_dataloader` from the datamodule are copied over to the model on .fit and .test without having to pass `trainer.fit(model, datamodule=datamodule)`.

Reply:
As per user feedback, I'm about to add implicit calls to prepare_data and setup of datamodule on .fit/.test if they haven't been called yet.